### PR TITLE
feat: Add relevant fields to the Explore Catalog CSV Download

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

- combine discovery metadata with algolia hits
- add new metadata to CSV download
- update tests

## References

- [ENT-5295](https://openedx.atlassian.net/browse/ENT-5295)

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/views.py [hunk 1]
```diff
@@ -317,13 +323,40 @@ def construct_csv_row(self, hit):
...
         skills = [skill['name'] for skill in hit.get('skills', [])]
         csv_row.append(', '.join(skills))
+
+        discovery_course = hit.get('discovery_course', {})
+        discovery_course_runs = discovery_course.get('course_runs', [])
+
+        try:
+           discovery_first_course_run = discovery_course_runs[0]
+        except IndexError:
+           discovery_first_course_run = {}
```
Comment (binodpant) [inline enterprise_catalog/apps/api/v1/views.py]: so if there's no course_runs we will still add a csv_row with null data?

File: enterprise_catalog/apps/api/v1/views.py [hunk 2]
```diff
@@ -343,7 +376,19 @@ def retrieve_indexed_data(self, facets, algoliaQuery):
         algolia_hits = []
         page = algolia_client.algolia_index.search(algoliaQuery, search_options)
         while len(page['hits']) > 0:
-            algolia_hits.extend(page['hits'])
+            course_keys_chunk = [hit['key'] for hit in page.get('hits', [])]
+            query_params = {'keys': ','.join(course_keys_chunk)}
+            courses = discovery_client.get_courses(query_params=query_params)
+            course_by_key = {}
+            for course in courses:
```
Comment (binodpant) [inline enterprise_catalog/apps/api/v1/views.py:386]: not sure I quite get this part and how this helps with the additional fields above...

File: enterprise_catalog/apps/api/v1/views.py [hunk 3]
```diff
@@ -343,7 +376,19 @@ def retrieve_indexed_data(self, facets, algoliaQuery):
...
+                course_by_key[course['key']] = course
+
+            for hit in page['hits']:
+                if course_by_key.get(hit['key']):
+                    hit['discovery_course'] = course_by_key.get(hit['key'])
+                algolia_hits.append(hit)
+
+            # algolia_hits.extend(page['hits'])
```
Comment (binodpant) [inline enterprise_catalog/apps/api/v1/views.py]: remove this commented line?

File: enterprise_catalog/apps/api/v1/views.py (hunk 1 above)
Comment (johnnagro) [reply enterprise_catalog/apps/api/v1/views.py]: Correct, that is my intention.

File: enterprise_catalog/apps/api/v1/views.py (hunk 2 above)
Comment (johnnagro) [reply enterprise_catalog/apps/api/v1/views.py:386]: the discovery api returns a list of courses, which i want to combine with the algolia course results. to improve performance, i am building a lookup dictionary of the discovery course metadata results, keyed by the course key. then when i iterate over the algolia results (hits) i can efficiently look in the dictionary to see if there is a matching (by course key) course discovery metadata result. i add the extra discovery data to the hit dictionary and append it to the results list.

does that make sense?

Comment (binodpant) [reply enterprise_catalog/apps/api/v1/views.py:386]: ah I see it’s an intermediate data structure, was trying to make sense of it. this clarifies it thanks!

**Review by binodpant — APPROVED** [review]
lgtm!

**Pull request merged**